src/main.ts

```diff
@@ -100,13 +103,43 @@ const executeDockerCommand = async (dockerCommand: string, continueOnError: bool
     var execOptions: any = {
         outStream: new NullOutstreamStringWritable({ decodeStrings: false }),
         listeners: {
-            stdout: (data: any) => console.log(data.toString()), //to log the script output while the script is running.
-            errline: (data: string) => {
+            stdout: async (data: any) => {
+                let scannedResult = {result: null};
+                if(!config.credScanEnable || !process.env.CREDSCAN){
+                    console.log(data.toString());
+                }
+                else if(config.credScanEnable || process.env.CREDSCAN){
+                    await cs.credscan(data.toString(), scannedResult);
+                    if(scannedResult.result){
+                        console.log(scannedResult.result);
+                    }
+                    else {
+                        console.log(data.toString());
+                    }
+                }
+                else console.log(data.toString());
+            }, //to log the script output while the script is running.
+            errline: async (data: string) => {
                 if (!shouldOutputErrorStream) {
                     errorStream += data + os.EOL;
                 }
                 else {
-                    console.log(data);
+                    let scannedResult = {result: null};
+                    if(!config.credScanEnable || !process.env.CREDSCAN){
+                        console.log(data);
+                    }
+                    else if(config.credScanEnable || process.env.CREDSCAN){

```

Comment:
do we really need this if condn?

Reply:
I changed the if conditions again so that if the feature flag is disabled then the user can't enable unless they enable the feature flag

Reply:
logic will reach L126 only when both config.credScanEnable, process.env.CREDSCAN are true. So no need of explicit if condition.

Reply:
I changed that part to: `if (!config.credScanEnable) {
        console.log(data);
    }
    else if (!process.env.CREDSCAN) {
        console.log(data);
    }
    else {
        await cs.credscan(data, scannedResult);
        if (scannedResult.result) {
            console.log(scannedResult.result);
        }
        else {
            console.log(data);
        }
    }`
Should I change this to 
`if (!config.credScanEnable || !process.env.CREDSCAN) {
        console.log(data);
    }
    else {
        await cs.credscan(data, scannedResult);
        if (scannedResult.result) {
            console.log(scannedResult.result);
        }
        else {
            console.log(data);
        }
    }`